diff -up evolution-ews-gnome-3-0.gitcc16df2/src/calendar/e-cal-backend-ews.c.free-busy-timezone-change evolution-ews-gnome-3-0.gitcc16df2/src/calendar/e-cal-backend-ews.c
--- evolution-ews-gnome-3-0.gitcc16df2/src/calendar/e-cal-backend-ews.c.free-busy-timezone-change	2016-08-31 12:42:56.478955173 +0200
+++ evolution-ews-gnome-3-0.gitcc16df2/src/calendar/e-cal-backend-ews.c	2016-08-31 12:42:56.495955173 +0200
@@ -4318,7 +4318,6 @@ typedef struct {
 	GSList *users;
 	time_t start;
 	time_t end;
-	icaltimezone *timezone;
 } EwsFreeBusyData;
 
 static void
@@ -4328,8 +4327,9 @@ prepare_free_busy_request (ESoapMessage 
 	EwsFreeBusyData *free_busy_data = user_data;
 	GSList *addr;
 	icaltimetype t_start, t_end;
+	icaltimezone *utc_zone = icaltimezone_get_utc_timezone ();
 
-	ewscal_set_availability_timezone (msg, free_busy_data->timezone);
+	ewscal_set_availability_timezone (msg, utc_zone);
 
 	e_soap_message_start_element (msg, "MailboxDataArray", "messages", NULL);
 
@@ -4351,8 +4351,8 @@ prepare_free_busy_request (ESoapMessage 
 	e_soap_message_start_element (msg, "FreeBusyViewOptions", NULL, NULL);
 
 	e_soap_message_start_element (msg, "TimeWindow", NULL, NULL);
-	t_start = icaltime_from_timet_with_zone (free_busy_data->start, 0, free_busy_data->timezone);
-	t_end = icaltime_from_timet_with_zone (free_busy_data->end, 0, free_busy_data->timezone);
+	t_start = icaltime_from_timet_with_zone (free_busy_data->start, 0, utc_zone);
+	t_end = icaltime_from_timet_with_zone (free_busy_data->end, 0, utc_zone);
 	ewscal_set_time (msg, "StartTime", &t_start, FALSE);
 	ewscal_set_time (msg, "EndTime", &t_end, FALSE);
 	e_soap_message_end_element (msg); /* "TimeWindow" */
@@ -4457,7 +4457,6 @@ e_cal_backend_ews_get_free_busy (ECalBac
 	free_busy_data->users = users_copy;
 	free_busy_data->start = start;
 	free_busy_data->end = end;
-	free_busy_data->timezone = priv->default_zone;
 
 	e_ews_connection_get_free_busy (
 		priv->cnc,
